tests/console/system_prepare.pm

```diff
@@ -27,6 +27,12 @@ use warnings;
 
 sub run {
     my ($self) = @_;
+
+    # We should call reconnect_mgmt_console to handle_grub_zvm/grab serial iucvconn etc for s390 backend
+    if (check_var("BACKEND", "s390x") && check_var("UPGRADE", "1") && is_sle('15+')) {
+        reconnect_mgmt_console;
```

Comment:
Don't we do this call in `reboot_after_installation` module? Looks like it belongs there

Reply:
Input logic into reboot_after_installation or current system_prepare.pm seems same to me. But i feel if we schedule reconnect_mgmt_console.pm module between reboot_after_installation and system_prepare.pm for this kind of case is better, and this way we do not need change any logic of any module. For example:

....
reboot_after_installation 
reconnect_mgmt_console.pm  <==== insert here by using yaml file to control schedule
post_upgrade
system_prepare
....

what's your opinion ?  @rwx788 @Soulofdestiny @okurz

Reply:
Indeed, I would say that it's better solution to just use `reconnect_mgmt_console` module for those scenarios, to at least try to keep module being atomic. Will that work for you?

Reply:
Sure, work for me.
Then i will add extra yaml schedule file with reconnect_mgmt_console.pm insert into it.  
@okurz what's your feedback on this solution? If ok then i will start working on this.
@Soulofdestiny  i suppose you also support this, correct?

Reply:
I agree that scheduling a module seems to be better than putting the reconnect in the test module "system_prepare". However in the long run I don't think the current handling in the YAML schedules will be very sustainable as it will not be clear why the reconnect test module is necessary or one will run into the same problem again when coming up with a new schedule. At best a user just calls `select_console …` and that does all what is necessary. Can we maybe call `reconnect_mgmt_console` as part of `activate_console` whenever it is needed, e.g. when a console has been previously reset?

Reply:
@coolgw works for me